### PR TITLE
1844 data polling bug & 6797 production vs develop reference to `hc-web-client connect()`

### PR DESCRIPTION
**Pull request by JettTech**

### CODE REVIEW

File: package.json
Comment (robbiecarlton) [inline package.json]: `conductor-config-manuals.toml` should be `conductor-config-manual.toml`

File: pkgs.nix
Comment (robbiecarlton) [inline pkgs.nix]: this bumps us to v39?

File: scripts/holochain-client.js
Comment (robbiecarlton) [inline scripts/holochain-client.js:11]: why remove this env var?

File: src/holochainClient.js
Comment (robbiecarlton) [inline src/holochainClient.js]: why replace with hardcoded version?

File: src/holofuel/components/SideMenu/SideMenu.js
Comment (robbiecarlton) [inline src/holofuel/components/SideMenu/SideMenu.js]: does this mean we can remove the `TOS` page entirely?

File: src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js
Comment (robbiecarlton) [inline src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js:20]: This did the trick?

It's definitely a bummer and we want to keep looking at other ways to do this as we want to get the benefit of caching.

**Review by robbiecarlton — CHANGES_REQUESTED** [review]

File: src/holochainClient.js
Comment (zo-el) [reply src/holochainClient.js]: Can we implement this in this way:
```
const url = process.env.NODE_ENV === 'production' ? undefined : process.env.REACT_APP_DNA_INTERFACE_URL
```

**Review by zo-el — CHANGES_REQUESTED** [review]
We need to deal with automating the `url` var

File: src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js
Comment (zo-el) [reply src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js:20]: @robbiecarlton Honestly, I don't see any other way other than this implementation from @JettTech. 
Because we need to pull this from the network.
Correct me if I am wrong, but the only way this is solved is when we implement `signals` in the Holofuel DNA so that we can use some sort of `subscription` method

File: src/holofuel/components/SideMenu/SideMenu.js
Comment (JettTech) [reply src/holofuel/components/SideMenu/SideMenu.js]: The TOS has been removed from the router, so I changed this link to a placeholder to avoid erroring out.  Has the design changed to remove this entirely? Do we have clarity on whether this will be a page internal to our app?  I would like to defer to the designs to answer this question.

File: src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js
Comment (JettTech) [reply src/holofuel/components/layout/PrimaryLayout/PrimaryLayout.js:20]: @robbiecarlton - True, as we discussed previously, forcing network calls is not ideal, but does work for now.  As we've also discussed, and as @zo-el mentioned here, I do think we can revisit and reform this approach once we implement signals and subscriptions.  That said, I am still open to implementing a more optimal solution, if you know one. 

@zo-el - I agree with the idea of optimizing our calls with subscriptions and signals once signals are implemented in the hf dna. Is there any movement toward this in Holo Central?

File: package.json
Comment (JettTech) [reply package.json]: Good catch! :)

File: pkgs.nix
Comment (JettTech) [reply pkgs.nix]: Yes, this was tied to v39 and is dependent on PR#129, so we should merge that first. :+1:

File: scripts/holochain-client.js
Comment (JettTech) [reply scripts/holochain-client.js:11]: After preparing our build for holoscape and building around recommended hc patterns, I highly recommend referencing logic that determines whether we're in a development or production environment and assigns the url value accordingly.  So, this is a cleanup task.  I will turn this into a logical statement. :+1:

File: src/holochainClient.js
Comment (JettTech) [reply src/holochainClient.js]: Same thought/reason as above.  I meant to clean up this hard-coded vars, so thanks for that catch!  @zo-el - I like the logical statement you proposed, and my intention is to implement this or something similar to leverage the benefit of a env var for testing, but following the 'hc-web-client' expected behavior for production environments for best coordination with the conductor and other hc apps.

File: src/holofuel/components/SideMenu/SideMenu.js
Comment (JettTech) [reply src/holofuel/components/SideMenu/SideMenu.js]: Ok. So I checked and it looks like TOS is still apart of the sidebar menu, but disclaimer is not and a link to 'Help' has been since added.  I made the updates accordingly. :+1:

File: src/holochainClient.js
Comment (robbiecarlton) [reply src/holochainClient.js]: this just seems redundant

`const url = process.env.REACT_APP_DNA_INTERFACE_URL`

is the same as your suggestion @zo-el. This is what env variables are for. Set the variable how you need it in the specific environment.

File: src/holochainClient.js
Comment (robbiecarlton) [inline src/holochainClient.js:37]: this is the same as `url: process.env.REACT_APP_DNA_INTERFACE_URL`

I don't understand why we're doing this. Just let the environment determine it.

**Review by robbiecarlton — APPROVED** [review]
I feel differently to you and Joel about specifying what the 'url' param should be in production.

It just removes flexibility and doesn't gain anything. What good thing comes from us restraining the value of the url when in a production environment?

That being said, don't let it block the PR.

File: src/holochainClient.js
Comment (zo-el) [reply src/holochainClient.js:37]: @robbiecarlton Yes, but this would mean that Everytime you have to make a build you will have to remove your env var from your .env file.
But if u use the NODE_ENV you do not have to deal with this issue. This helps in automation.
Especially when u have to run test as well as make a build, which we are doing on Hydra.

**Comment by robbiecarlton** [comment]
we can handle that by having separate .env files for production and
development

**Comment by zo-el** [comment]
I don't think that is standard. Also it doesn't solve issues with CI's

**Comment by robbiecarlton** [comment]
It’s definitely normal practice. it’s built into create-react-app. And I
think you can have one for .env.test for CI

https://dev.to/jam3/managing-env-variables-for-provisional-builds-h37

**Comment by zo-el** [comment]
I think the CI should not be testing things out on different environment settings. 
I think the testing should be done in a `development` setting and the build should be done in `production` settings.
Also in CI's, we do not have .env files. its easier to just set the env var without the `.env`, which makes using `.env.test` not really optimal.
I did look through the above link and I don't see a need for the third environment, because there is nothing special about the CI environment. And we would need another dependency to implement it.
Also, we solve all of these issues just by using `NODE_ENV`. 
The `NODE_ENV` is used in create_react_app anyway in the react-scripts.

Is there a reason why you prefer not using this variable? 
That would probably clear things out for me.

**Comment by robbiecarlton** [comment]
It’s just redundant. The whole point of an env variable is to set the
variable to what you need in a specific environment. This is a canonical
use case of an env variable. So instead we’re just jury rigging an env
variable when we could be using the mechanism which is designed for this
purpose.

So for CI we set the variable to whatever we need, in production we set it
and in development we set it. Done. The code doesn’t care because the code
doesn’t need to know.

**Comment by zo-el** [comment]
I don't understand what is redundant. Can you explain?

**Review by zo-el — APPROVED** [review]

**Pull request merged**
